src/uct/base/uct_iface.h

```diff
@@ -449,10 +463,10 @@ typedef struct uct_iface_local_addr_ns {
  * @param _cfg_table      Transport configuration table
  * @param _cfg_struct     Struct type defining transport configuration
  */
-#define UCT_TL_REGISTER_DEF(_component, _name, _query_devices, _iface_class, \
-                            _cfg_prefix, _cfg_table, _cfg_struct) \
+#define UCT_TL_DEF(_component, _name, _query_devices, _iface_class, \
+                   _cfg_prefix, _cfg_table, _cfg_struct) \
     \
-    static uct_tl_t uct_##_name##_tl = { \
+    uct_tl_t UCT_TL_NAME(_name) = { \
```

Comment:
can we separate the part that defines the struct fields (lines 469..479) to another macro, and then use this macro to define IB transports such as rc,ud,...?

why - seems that it's enough to keep the array of UCT_TL_NAME (uct_ib_tls), and during IB initialization go over this array. so for IB transports, we don't need to define ctor/dtor functions.